src/agni/server/receiver/InfoRequestReceiver.java

```diff
@@ -1,17 +1,41 @@
 package agni.server.receiver;
 
+import java.nio.ByteBuffer;
+import java.util.Vector;
+
 public class InfoRequestReceiver implements MessageParser {
-    private InfoListener[] infoListeners;
+    private Vector <InfoListener> infoListeners = null;
 
     public InfoRequestReceiver() {
-
+        infoListeners = new Vector<InfoListener>();
     }
 
-    private void notifyInfoRequest() {
+    private void notifyInfoRequest(String ip, byte type) {
```

Comment:
Probably in this PR we should be translating the `type` from type `byte` into some form of enum. Code for the enum would be:

```java
    public enum InfoType {
        SERVER_IP((byte)0x00),
        SERVER_PORT((byte)0x01),
        SERVER_NAME((byte)0x02);
        CURRENT_ONLINE((byte)0x03);
        CURRENT_CHATS((byte)0x04);

        private final byte bytes;
        private Status(byte bytes) {
            this.bytes = bytes;
        }

        public byte bytes() {
            return bytes;
        }
    }
```